src/twisted/test/test_defer.py

```diff
@@ -1735,6 +1736,61 @@ def test_fromCoroutineRequiresCoroutine(self) -> None:
         for thing in thingsThatAreNotCoroutines:
             self.assertRaises(defer.NotACoroutineError, Deferred.fromCoroutine, thing)
 
+    def test_inlineCallbacksCancelCaptured(self) -> None:
```

Comment:
Just asking

Is this test related to circular references?

Was this test created as there was regression that you discovered while refactoring the code, and the bug was not detected by the previous set of tests?

--------

I know that the existing code is already not idea.

Maybe, as part of this PR, we can create a new `InlineCallbacksTests` class that will group all the tests targeting `inlineCallbacsk` usage

The current `DeferredTests` is a mess, as there are so many tests here and is hard to check for duplicate tests.

Reply:
The test isn't related to circular references, nor really a regression test. It's a new test that covers this particular edge case of cancellations that weren't covered by any other tests. It's a fiddly bit of logic, so didn't want to refactor it without more coverage.

---

Happy to move the tests into their own class. I'll do that now.

Reply:
Done.